src/test/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceTest.java

```diff
@@ -106,6 +108,165 @@ class JurorAppearanceServiceTest {
+
+    }
+
+    @Test
+    void addAttendanceDayJurorPoolNotFound() {
+        Juror juror = new Juror();
+        juror.setJurorNumber(JUROR_123456789);
+
+        PoolRequest poolRequest = new PoolRequest();
+        poolRequest.setPoolNumber("123456789");
+
+        JurorPool jurorPool = getJurorPool(juror, IJurorStatus.RESPONDED);
+        jurorPool.setPool(poolRequest);
+
+        doReturn(jurorPool).when(jurorPoolRepository)
+            .findByJurorJurorNumberAndPoolPoolNumber(
+                JUROR_123456789, "111111111");
+
+        AddAttendanceDayDto dto = buildAddAttendanceDayDto();
+
+        assertThatExceptionOfType(MojException.NotFound.class).isThrownBy(() ->
+                jurorAppearanceService.addAttendanceDay(buildPayload("417", List.of("417")),
+                    dto)).as("No valid juror pool found")
+            .withMessageContaining("No valid juror pool found");
+
+    }
+
+    @Test
+    void processAppearanceBooleanTrue() {
+        Juror juror = new Juror();
+        juror.setJurorNumber(JUROR_123456789);
+
+        PoolRequest poolRequest = new PoolRequest();
+        poolRequest.setPoolNumber("123456789");
+        JurorPool jurorPool = getJurorPool(juror, IJurorStatus.RESPONDED);
+        jurorPool.setPool(poolRequest);
+        juror.setAssociatedPools(Collections.singleton(jurorPool));
+        CourtLocation courtLocation = getCourtLocation();
+
+        doReturn(Optional.of(juror)).when(jurorRepository).findById(JUROR_123456789);
+        doReturn(Collections.singletonList(jurorPool)).when(jurorPoolRepository)
+            .findByJurorJurorNumberAndIsActiveOrderByPoolReturnDateDesc(
+                JUROR_123456789, true);
+        doReturn(Optional.of(courtLocation)).when(courtLocationRepository).findById(anyString());
+
+        JurorAppearanceResponseDto.JurorAppearanceResponseData appearanceData = JurorAppearanceResponseDto
+            .JurorAppearanceResponseData.builder().jurorNumber(JUROR_123456789)
+            .lastName("LASTNAME")
+            .firstName("FIRSTNAME")
+            .checkInTime(LocalTime.of(9, 30))
+            .jurorStatus(IJurorStatus.RESPONDED)
+            .build();
+
+        List<JurorAppearanceResponseDto.JurorAppearanceResponseData> appearanceDataList = new ArrayList<>();
+        appearanceDataList.add(appearanceData);
+
+        when(appearanceRepository.getAppearanceRecords(anyString(), any(), anyString()))
+            .thenReturn(appearanceDataList);
+
+        JurorAppearanceDto jurorAppearanceDto = buildJurorAppearanceDto();
+        jurorAppearanceDto.setCheckOutTime(LocalTime.of(17, 30));
+
+        jurorAppearanceService.processAppearance(buildPayload(OWNER_415, Arrays.asList("415", "462", "767")),
+            jurorAppearanceDto, true);
+
+        verify(jurorRepository, times(1))
+            .findById(JUROR_123456789);
+        verify(jurorPoolRepository, times(1))
+            .findByJurorJurorNumberAndIsActiveOrderByPoolReturnDateDesc(any(), anyBoolean());
```

Comment:
Should not use any in verify calls unless you are verify it was never called

Reply:
Updated